internal/juju/secrets.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the Apache License, Version 2.0, see LICENCE file for details.
+
+package juju
+
+import (
+	"errors"
+
+	"github.com/juju/juju/api"
+	apisecrets "github.com/juju/juju/api/client/secrets"
+	coresecrets "github.com/juju/juju/core/secrets"
+)
+
+type secretsClient struct {
+	SharedClient
+
+	getSecretAPIClient func(connection api.Connection) SecretAPIClient
+}
+
+type CreateSecretInput struct {
+	ModelName string
+	Name      string
+	Value     map[string]string
+	Info      string
+}
+
+type CreateSecretOutput struct {
+	SecretId string
+}
+
+type ReadSecretInput struct {
+	ModelName string
+	Name      string
+}
+
+type ReadSecretOutput struct {
+	Value map[string]string
+}
+
+type UpdateSecretInput struct {
+	ModelName string
+	Name      string
+	Value     map[string]string
+	Info      string
+}
+
+type DeleteSecretInput struct {
+	ModelName string
+	Name      string
+}
+
+func newSecretsClient(sc SharedClient) *secretsClient {
+	return &secretsClient{
+		SharedClient: sc,
+		getSecretAPIClient: func(connection api.Connection) SecretAPIClient {
+			return apisecrets.NewClient(connection)
+		},
+	}
+}
+
+// CreateSecret creates a new secret.
+func (c *secretsClient) CreateSecret(input *CreateSecretInput) (*CreateSecretOutput, error) {
```

Comment:
For the Output structures, let's use the struct rather than a pointer. There will no be modified passing to other methods, nor are they big enough that copies are not okay.